fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/api/LoanTransactionsApiResource.java

```diff
@@ -637,4 +729,19 @@ private Long getResolvedLoanId(final Long loanId, final ExternalId loanExternalI
         }
         return resolvedLoanId;
     }
+
+    private Long getResolvedLoanIdWithExistsCheck(final Long loanId, final ExternalId loanExternalId) {
+        if (loanId != null) {
+            loanReadPlatformService.retrieveOne(loanId);
```

Comment:
Would you mind to rework this? We dont want to fetch loan and map a loan data for no reason... :/

if you want to check whether it is existing, do with exists query or do it at a later stage once you fetch the loan...

Reply:
Reworked with exists query.